Simplify default serializer deprecation docs (#545)
ember-learn/deprecation-app: content/ember-data/v3/default-serializer.md

---
## Default serializer deprecations

### Deprecate adapter.defaultSerializer fallback

Moving forward, every app or addon that uses Ember Data must have a serializer explicitly defined.

> Previously, if no application or model-type-specific serializer was specified, the store would attempt to lookup a serializer via the `defaultSerializer` property on the type's adapter. This behavior is deprecated in favor of explicitly defining a type-specific serializer or application serializer.

Reviewer: ```suggestion
Previously, if no application or model-type-specific serializer was specified, the store would attempt to look up a serializer via the `defaultSerializer` property on the type's adapter. This behavior is deprecated in favor of explicitly defining a model-type-specific serializer or application serializer.
```

> ### deprecate adapter.defaultSerializer fallback

Reviewer 2: ```suggestion
### Deprecate adapter.defaultSerializer fallback
```

- `RESTAdapter`: `-rest` (`@ember-data/serializer/rest`)
- `JSONAPIAdapter`: `-json-api` (`@ember-data/serializer/json-api`)

### Clearing these deprecations

If all the adapters in your app are the same kind (such as JSONAPI, REST, or JSON), you should create an
application serializer to match.

> For example, if you use only JSONAPI adapters, creating the file below in `app/serializers/application.js`

Reviewer: ```suggestion
For example, if you use only JSONAPI adapters, creating the following application serializer
```

will resolve the deprecation:

```javascript

> // app/serializers/application.js

Reviewer: ```suggestion
```javascript {data-filename=app/serializers/application.js
```

> export { default } from '@ember-data/serializer/json-api';

Reviewer: Were these module paths available in 3.12?

Reviewer 2: @locks i think so 🤷‍♀ 
https://github.com/emberjs/data/blob/e0732481885abde4eb187d3a1573276374068a2f/packages/serializer/addon/json-api.js#L412

Reviewer: https://guides.emberjs.com/v3.14.0/models/defining-models/
We seem to only have switched the defaults in 3.15 though. Should we align these with the default at the time of the deprecation?

Reviewer 2: yeah makes sense

```

> If your app uses different adapter types for different models, you should make one serializer for each model type. For example, if a certain model uses the `RESTAdapter`, create an `app/serializers/[model-type].js` file with the following:

Reviewer: What if you give a concrete model name as an example and use the `data-filename` facility to show the path?

```js

>     import RESTSerializer from '@ember-data/serializer/rest';

Reviewer: This is pre-existent, but I don't think it should have indentation.

    }
```

> More information about custom serializers can be found in the [Serializer API Docs](https://api.emberjs.com/ember-data/release/modules/@ember-data%2Fserializer) or on the [ember.js/guides](https://guides.emberjs.com/release/models/customizing-serializers/#toc_customizing-serializers)

Reviewer: ```suggestion
More information about custom serializers can be found in the [Serializer API Docs](https://api.emberjs.com/ember-data/release/modules/@ember-data%2Fserializer) or on the [official guides](https://guides.emberjs.com/release/models/customizing-serializers/#toc_customizing-serializers)
```

Reviewer: English question, does anyone have a good mnemonic for "on the guides" vs "in the guides"?

Reviewer 2: I think I prefer in, but don't have super strong opinions. Probably would make more sense to use on if were saying `on the Guides website`, but either way I think it def would sound better if we unified in/on b/w the api docs and guides here.
